solidity/contracts/middleware/InterchainQueryRouter.sol

```diff
@@ -140,27 +88,27 @@ contract InterchainQueryRouter is Router, IInterchainQueryRouter {
         bytes32, // router sender
         bytes calldata _message
     ) internal override {
-        Action action = Action(uint8(bytes1(_message[31])));
-        if (action == Action.DISPATCH) {
-            (
-                ,
-                address sender,
-                CallLib.Call[] memory calls,
-                bytes[] memory callbacks
-            ) = abi.decode(
-                    _message,
-                    (Action, address, CallLib.Call[], bytes[])
-                );
-            callbacks = calls._multicallAndResolve(callbacks);
-            _dispatch(_origin, abi.encode(Action.RESOLVE, sender, callbacks));
+        InterchainCallMessage.CallType calltype = InterchainCallMessage
+            .calltype(_message);
+        bytes32 sender = InterchainCallMessage.sender(_message);
+        if (
+            calltype == InterchainCallMessage.CallType.STATIC_CALL_WITH_CALLBACK
+        ) {
             emit QueryReturned(_origin, sender);
-        } else if (action == Action.RESOLVE) {
-            (, address sender, bytes[] memory resolveCallbacks) = abi.decode(
-                _message,
-                (Action, address, bytes[])
+            CallLib.StaticCallWithCallback[]
```

Comment:
Should you emit after executing the multistaticcall?

Reply:
if I do this, slither correctly flags that the arbitrary calls out can cause events to be emitted out of order
probably fine but I will have to disable that warning